### ISSUE TITLE
Limit results by radius around lat/lon

### ISSUE THREAD

**Issue by RamRaider** [datagovuk/apiserver#77]
It appears that ALL results are returned from queries such as 

> //data.gov.uk/data/api/service/health/hospitals/nearest?lat=52.64105830207872&lon=1.032092571258545

 which seems pointless as the returned data is much larger than it needs to be. Is there or can there be a distance limiting parameter - ie 

> //data.gov.uk/data/api/service/health/hospitals/nearest?lat=52.64105830207872&lon=1.032092571258545&radius=20

 etc?? 

In the API documentation it states there is the ability to query the database directly by crafting the sql and appending to the endpoint url - but it is unclear how one would use this approach to achieve the desired distance limit mentioned.

**Comment by rossjones** [comment]
Yes, limiting and radius and paging are all things that need doing.  There's no support for radius at present, although *theoretically* it's just a case of using ST_Distance_Sphere in this case it's a little more complicated.

In the meantime, so you can keep your resultsets small you could go to  ```https://data.gov.uk/data/api/health#sql``` and try the following query

```sql
select * from hospitals 
order by ST_Distance(LatLong, ST_SetSRID(ST_MakePoint(52.64105830207872, 1.032092571258545), 4326))
limit 10;
```

If you then click download as json you should have the URL to use.
Obviously it isn't as useful as radius, but it might help you progress in the meantime.

**Comment by RamRaider** [comment]
:) Brilliant - thanks for the useful response and code! I shall give the example sql a try and see how I get on with that approach

**Comment by RamRaider** [comment]
I have attempted to implement the suggested method and for some of the endpoints it is perfect ( hospitals, coach, ferries and trains etc ) but for the other health services such as GP surgeries, clinics, pharmacies it returns an error stating 

> {"success":false,"error":"Query would take too long to run, please apply a LIMIT of less than 5000 rows"}

This occurs when the query has a limit of 10, ie:

> https://data.gov.uk/data/api/service/health/sql?query=SELECT * FROM clinics ORDER BY ST_Distance(LatLong, ST_SetSRID(ST_MakePoint(52.64155308408907,0.9916448593139648),4326)) LIMIT 10

I tried with a limit of 1 and briefly a recordset was displayed before reverting to the above error message - please advise how I might rectify the situation.
